### PR TITLE
Add stub for getting client-specific connection

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

To make the next PR smaller.

## Readiness checklist

* [ ] I added unit tests for new functionality or bug fixes.
* [ ] I added integration tests for new functionality or bug fixes.
* [ ] I added compatibility tests for new functionality or bug fixes.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/pg/msg_insert.go [hunk 1]
```diff
@@ -128,7 +132,7 @@ func (h *Handler) insertMany(ctx context.Context, sp *pgdb.SQLParam, docs *types
 }
 
 // insert prepares and executes actual INSERT request to Postgres.
-func (h *Handler) insert(ctx context.Context, sp *pgdb.SQLParam, doc any) error {
+func (h *Handler) insert(ctx context.Context, dbPool *pgdb.Pool, sp *pgdb.SQLParam, doc any) error {
```
Comment (rumyantseva) [inline internal/handlers/pg/msg_insert.go:135]: If we pass `dbPool` as param, should method `func (h *Handler) insert` be just `func insert` then?

File: internal/handlers/pg/pg.go [hunk 2]
```diff
@@ -41,13 +48,45 @@ type NewOpts struct {
...
+
+	for k, pgPool := range h.pools {
+		pgPool.Close()
+		delete(h.pools, k)
+	}
+}
+
+// DBPool returns database connection pool for the given client connection.
+func (h *Handler) DBPool(ctx context.Context) (*pgdb.Pool, error) {
+	// TODO make real implementation; the current one is a stub
```
Comment (rumyantseva) [inline internal/handlers/pg/pg.go:69]: I understand that this is not for the current PR, as the current PR is a stub.
As no issue is mentioned, I expect it to be fixed in the next PR (that we start passing auth data from the client).

File: internal/handlers/pg/pg.go [hunk 3]
```diff
@@ -16,22 +16,29 @@
...
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/state"
 )
 
 // Handler implements handlers.Interface on top of PostgreSQL.
 type Handler struct {
 	*NewOpts
+
+	rw    sync.RWMutex
+	pools map[string]*pgdb.Pool
```
Comment (rumyantseva) [inline internal/handlers/pg/pg.go:36]: The idea is that none of `Msg` methods should request this map directly, they should always call `h.DBPool(ctx)` instead. However, in principle such call is possible. Should we leave a comment that it is expected to use `h.DBPool`, but not this map + mutex directly? Or is it obvious?

**Review by rumyantseva — COMMENTED** [review]
Looks good to me overall, left a couple of comments/questions.

File: internal/handlers/pg/pg.go (hunk 2 above)
Comment (AlekSi) [reply internal/handlers/pg/pg.go:69]: #1725

File: internal/handlers/pg/pg.go (hunk 3 above)
Comment (AlekSi) [reply internal/handlers/pg/pg.go:36]: I will add a comment in #1725, but someone should _really_ try to use that map…

File: internal/handlers/pg/msg_insert.go (hunk 1 above)
Comment (AlekSi) [reply internal/handlers/pg/msg_insert.go:135]: Good point, will do in the next PR

**Review by rumyantseva — APPROVED** [review]
LGTM (with some expected additions in the next PR).

:shipit:

**Comment by noisersup** [comment]
I was trying to review it _before merge_, but I don't understand the point of this. By client-specific we mean the connections from FerretDB to PostgreSQL, right? Do we have plans to support multiple postgresql connections in parallel here? Sorry if the answer is obvious but I feel lost here.

**Comment by AlekSi** [comment]
@noisersup #1725 might make it clear

**Pull request merged**